performance/query_processing.rst

```diff
@@ -53,6 +63,179 @@ The task tracker executor is well suited for long running, complex data warehous
+    SELECT page_id
+    FROM visits
+    GROUP BY page_id
+    ORDER BY count(*) DESC
+    LIMIT 20
+  )
+  GROUP BY page_id;
+
+::
+
+  GroupAggregate  (cost=0.00..0.00 rows=0 width=0)
+    Group Key: remote_scan.page_id
+    ->  Sort  (cost=0.00..0.00 rows=0 width=0)
+      Sort Key: remote_scan.page_id
+
+So far no change. The coordinator processes the final resultset in the same way.
+
+::
+
+      ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
+        ->  Distributed Subplan 6_1
+  .
+
+But here is something new. The custom scan has two large sub-trees, starting with this "distributed subplan."
+
+::
+
+          ->  Limit  (cost=0.00..0.00 rows=0 width=0)
+            ->  Sort  (cost=0.00..0.00 rows=0 width=0)
+              Sort Key: COALESCE((pg_catalog.sum((COALESCE((pg_catalog.sum(remote_scan.worker_column_2))::bigint, '0'::bigint))))::bigint, '0'::bigint) DESC
+              ->  HashAggregate  (cost=0.00..0.00 rows=0 width=0)
+                Group Key: remote_scan.page_id
+                ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
+                  Task Count: 32
+                  Tasks Shown: One of 32
+                  ->  Task
+                    Node: host=localhost port=5433 dbname=postgres
+                    ->  Limit  (cost=1883.00..1883.05 rows=20 width=12)
+                      ->  Sort  (cost=1883.00..1965.54 rows=33017 width=12)
+                        Sort Key: (count(*)) DESC
+                        ->  HashAggregate  (cost=674.25..1004.42 rows=33017 width=12)
+                          Group Key: page_id
+                          ->  Seq Scan on visits_102264 visits  (cost=0.00..509.17 rows=33017 width=4)
+  .
+
+This big chunk governs running the subquery and making its results available.
+
+::
+
+        Task Count: 32
+        Tasks Shown: One of 32
+        ->  Task
+          Node: host=localhost port=5433 dbname=postgres
+          ->  HashAggregate  (cost=734.53..899.61 rows=16508 width=8)
+            Group Key: visits.page_id, visits.session_id
+            ->  Hash Join  (cost=17.00..651.99 rows=16508 width=8)
+  .
+
+Like in the simpler query, the database will do a join within each worker, but this time not with the result of an ANY expression...
+
+::
+
+              Hash Cond: (visits.page_id = intermediate_result.page_id)
+  .
+
+... This time it joins against "intermediate results."
+
+::
+
+              ->  Seq Scan on visits_102264 visits  (cost=0.00..509.17 rows=33017 width=8)
+              ->  Hash  (cost=14.50..14.50 rows=200 width=4)
+                ->  HashAggregate  (cost=12.50..14.50 rows=200 width=4)
+                  Group Key: intermediate_result.page_id
+                  ->  Function Scan on read_intermediate_result intermediate_result  (cost=0.00..10.00 rows=1000 width=4)
+  .
+
+The intermediate results are internally retrieved from a function which loads data from a file that was passed between the worker nodes.
```

Comment:
Actually, the data from the first query is retrieved back to the master and then pushed back down to all workers. They don't transfer between the workers. Could you note that?